src/ert/_c_wrappers/enkf/ensemble_config.py

```diff
@@ -246,21 +247,29 @@ def gen_data_node(gen_data: List[str]) -> Optional[GenDataConfig]:
         report_steps = rangestring_to_list(options.get(ConfigKeys.REPORT_STEPS, ""))
 
         if os.path.isabs(res_file) or "%d" not in res_file:
-            logger.error(
-                f"The RESULT_FILE:{res_file} setting for {name} is invalid - "
-                "must have an embedded %d - and be a relative path"
+            result_file_context: ContextValue = next(
+                x for x in gen_data if x.startswith("RESULT_FILE:")
             )
-        elif not report_steps:
-            logger.error(
-                "The GEN_DATA keywords must have a REPORT_STEPS:xxxx defined"
-                "Several report steps separated with ',' and ranges with '-'"
-                "can be listed"
+            raise ConfigValidationError.from_info(
+                ErrorInfo(
+                    filename=result_file_context.token.filename,
+                    message=f"The RESULT_FILE:{res_file} setting for {name} is "
+                    f"invalid - must have an embedded %d - and be a relative path",
```

Comment:
just a litte suggestion regarding punctuation, you can skip this change if you want:
```suggestion
                    f"invalid - must have an embedded %d and be a relative path",
```